packages/shared/src/components/streak/popup/ReadingStreakPopup.tsx

```diff
@@ -33,52 +60,52 @@ export function ReadingStreakPopup({
   fullWidth,
 }: ReadingStreakPopupProps): ReactElement {
   const { user } = useAuthContext();
+  const [today, setToday] = useState(new Date());
   const { data: history } = useQuery(
     generateQueryKey(RequestKey.ReadingStreak30Days, user),
     () => getReadingStreak30Days(user.id),
     { staleTime: StaleTime.Default },
   );
 
-  const streaks = useMemo(() => {
-    return streakDays.map((value) => {
-      const day = value.getDay();
-      const date = value.getDate();
-      const isFuture = value > today;
-      const isCompleted =
-        !isFuture &&
-        history?.some(({ date: historyDate, reads }) => {
-          const dateToCompare = new Date(historyDate);
-          const sameDate = isSameDay(dateToCompare, value);
-
-          return sameDate && reads > 0;
-        });
-
-      const getStreak = () => {
-        if (isCompleted) {
-          return Streak.Completed;
-        }
+  // only change today and streakDays if the current day changed to the next one
+  if (!isSameDay(new Date(), today)) {
+    setToday(new Date());
+  }
+  const { dateToday, streakDays } = useMemo(
+    () => getStreakDays(today),
+    [today],
+  );
```

Comment:
Also in addition to https://github.com/dailydotdev/apps/pull/3075/files#r1589621107, we can also move this whole evaluation to below `useMemo`, no need to keep it separate since both memoizations have the same dependency `today.`